### PR TITLE
Added geo checks for uris in YAML

### PR DESCRIPTION
**Pull request by m-mayran**

### CODE REVIEW

File: tests/test_spawner.py [hunk 1]
```diff
@@ -589,8 +592,11 @@ def test_clustername(*args, **kwargs):
 
     config_built = spawner._build_cluster_config()
 
-    subnet_region = config_built['config']['gce_cluster_config']['subnetwork_uri'].split('/')[-3]
-    assert subnet_region == spawner.region
+    assert config_built['config']['gce_cluster_config']['subnetwork_uri'].split('/')[-3] == spawner.region
+    assert config_built['config']['master_config']['machine_type_uri'].split('/')[-3] == spawner.zone
```
Comment (JerryLeiDing) [inline tests/test_spawner.py:596]: This seems to be checking the wrong thing: shouldn't we assert that the machine type URI zone matches the zone the user selection to spawn the cluster in?

File: dataprocspawner/spawner.py [hunk 2]
```diff
@@ -929,7 +931,32 @@ def _build_cluster_config(self, cluster_data=None):
...
+      if server_group in cluster_data['config']:
+        # We do not check the zone because the user form overwrites it.
+        # if 'zone_uri' in cluster_data['config']['gce_cluster_config']:
+        #   self._check_uri_geo(
+        #     uri=cluster_data['config']['gce_cluster_config']['zone_uri'],
+        #     uri_geo_slice=-1,
+        #     expected_geo=self.zone
+        #   )
+        # Machine types must be in the same zone as the Dataproc Cluster.
+        if 'machine_type_uri' in cluster_data['config'][server_group]:
```
Comment (JerryLeiDing) [inline dataprocspawner/spawner.py:945]: Could we strip the zone from these entirely? That seems friendlier than forcing the config to match the spawned zone.

Comment (m-mayran) [reply dataprocspawner/spawner.py:945]: I thought of that actually. 
For example, replacing (https://www.googleapis.com/compute/v1/)projects/[projectId]/zones/us-east1-a/machineTypes/n1-standard-2 with n1-standar-2 every time?

File: tests/test_spawner.py (hunk 1 above)
Comment (m-mayran) [reply tests/test_spawner.py]: Good catch. Update in https://github.com/GoogleCloudDataproc/jupyterhub-dataprocspawner/pull/14/commits/d017eee3849e39413928b1b4a5ae2e124537f836

File: dataprocspawner/spawner.py (hunk 2 above)
Comment (JerryLeiDing) [reply dataprocspawner/spawner.py:945]: Yep, and the same thing with accelerators.

Comment (m-mayran) [reply dataprocspawner/spawner.py:940]: Done in https://github.com/GoogleCloudDataproc/jupyterhub-dataprocspawner/pull/14/commits/47dd837a10e7ef8d1abe001a81be76fef56dae87

**Pull request merged**